src/Provider/AbstractProvider.php

```diff
@@ -361,6 +364,9 @@ public function getResponse(RequestInterface $request)
 
             $response = (string) $httpResponse->getBody();
         } catch (HttpAdapterException $e) {
+            if (!$e->hasResponse()) {
+                throw $e;
```

Comment:
@SammyK do you think this is alright, or should we wrap the exception with a League version? It seems unnecessary to me, since this represents something the provider can't recover from.